src/main/java/com/faforever/api/data/domain/LegacyAccessLevel.java

```diff
@@ -0,0 +1,31 @@
+package com.faforever.api.data.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.springframework.security.core.GrantedAuthority;
+
+@Getter
+@AllArgsConstructor
+@Deprecated
+public enum LegacyAccessLevel implements GrantedAuthority {
+  ROLE_USER(0),
+  ROLE_MODERATOR(1),
+  ROLE_ADMINISTRATOR(2);
+
+  private final int code;
```

Comment:
If we intend to always keep the code like an array index, we could dismiss this and just use the enum's  ordinal.
If the code could be an arbitrary number, then we should keep it.

Reply:
It's not intended as an array index. It just reflects the current values for these roles in the database (which are very unconvenient because you can't distinguish any further.